### PR TITLE
Updates for production historical file processing

### PR DESCRIPTION
**Pull request by philbudne**
* Increase fetcher and parser container counts.
* Add --cleanup option to re-start orphaned files
* Use SQLite3 for local tracking DB: easier to view, back up and modify
* Handle CSV date/times without fractional seconds

### CODE REVIEW

File: indexer/tracker.py [hunk 1]
```diff
@@ -6,28 +6,43 @@
...
+class _TrackerException(AppException):
+    """
+    base class for FileTracker exceptions
+    """
+
+
+class TrackerStatus(_TrackerException):
     """
     thrown when file is in some state other than NOT_STARTED
     """
```
Comment (kilemensi) [inline indexer/tracker.py:36]: May be `StatusError`? TrackerStatus doesn't sound like an exception.

File: indexer/tracker.py [hunk 2]
```diff
@@ -6,28 +6,43 @@
...
 
 
+class _TrackerRetry(_TrackerException):
```
Comment (kilemensi) [inline indexer/tracker.py:39]: `_TrackerLocked`? Choice of retrying or giving up is up to the tracker itself I guess.

File: indexer/tracker.py [hunk 3]
```diff
@@ -72,98 +88,223 @@ def basename(self, fname: str) -> str:
...
+
+    def _start_if_available(self) -> None:
+        try:
+            self._retry_open_and_lock()
+            status, ts = self._db_get_status_time()
+            logger.debug("_start_if_available %s %d", status, ts)
+            if status == FileStatus.NOT_STARTED.name or (
+                self.cleanup
+                and status == FileStatus.STARTED.name
+                and (time.time() - ts) > OLD_SECONDS
```
Comment (kilemensi) [inline indexer/tracker.py:152]: 🤔 ... so we **won't** delete files that are less than 1 day old even if with `--cleanup`?

File: indexer/tracker.py [hunk 4]
```diff
@@ -72,98 +88,223 @@ def basename(self, fname: str) -> str:
...
+        GDBM and SQLite3 can both raise exceptions when DB locked
+        """
+        for sec in (0.0625, 0.125, 0.25, 0.5, 1.0, 2.0, 4.0, 8.0, 16.0):
+            try:
+                self._db_open_and_lock()
+                return
+            except _TrackerRetry:
+                logger.debug("_retry_open_and_lock sleeping %g", sec)
+                time.sleep(sec)
+        raise TrackerStatus("locked")
```
Comment (kilemensi) [inline indexer/tracker.py:179]: Nitty but isn't `TrackerStatus` tied to the "internal" state of a file i.e. FileStatus (NOT-STARTED, STATED and FINISHED)?

"external" state could be anything filesystem-related; file is locked, non-existing/invalid db path used, etc.

**Review by kilemensi — APPROVED** [review]
🚀

**Review by pgulley — APPROVED** [review]

**Review by thepsalmist — APPROVED** [review]

File: indexer/tracker.py (hunk 1 above)
Comment (philbudne) [reply indexer/tracker.py:36]: Replaced with `TrackerNotStartable`

File: indexer/tracker.py (hunk 2 above)
Comment (philbudne) [reply indexer/tracker.py:39]: replaced with `TrackerLocked` (exposed to user, see below)

File: indexer/tracker.py (hunk 3 above)
Comment (philbudne) [reply indexer/tracker.py:152]: It won't try to re-run files that were started (but not finished) until a day after they were marked as started

File: indexer/tracker.py (hunk 4 above)
Comment (philbudne) [reply indexer/tracker.py:179]: Will now (re)raise TrackerLocked to user when file cannot be locked and written to.

**Pull request merged**